### PR TITLE
Jupyterlab 4 upgrade

### PR DESCRIPTION
**Pull request by peytondmurray**
This PR ports the extension to JupyterLab 4, updates dependencies to their latest version (except for the ones that are shared/constrained by JupyterLab itself, which are matched to https://github.com/jupyterlab/jupyterlab/blob/master/package.json)

Here's a screenshot of the extension working with `jupyterlab==4.0.0a37`. I tried changing the output truncation limits and everything seemed okay.

image

The only possible issue I saw was a few warnings in the browser console about something related to `notebooklspadapter.ts`, which I don't think is related.

### CODE REVIEW

**Review by mlucool — COMMENTED** [review]
Thanks! This looks good.

Let's:
1. Update the CHANGELOG to indicate this change
2. Update package.json to 2.0.0

**Comment by peytondmurray** [comment]
Thanks for the input - the changes you asked for have been implemented. I'll rerequest review - can you make sure the `CHANGELOG` is okay?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## [2.0.0](https://github.com/deshaw/jupyterlab-limit-output/compare/v1.0.1...v2.0.0) (2023-04-04)
+
+### Changed
+
+- Ported to JupyterLab 4.
```
Comment (mlucool) [inline CHANGELOG.md]: Let's mark this as breaking (e.g. similar to https://github.com/deshaw/jupyterlab-execute-time/blob/master/CHANGELOG.md#200-2021-01-19)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+## [2.0.0](https://github.com/deshaw/jupyterlab-limit-output/compare/v1.0.1...v2.0.0) (2023-04-04)
```
Comment (mlucool) [inline CHANGELOG.md]: For release date `(UNRELEASED)` is correct until we choose to release. I suspect it doesn't make sense to release before lab is finalized.

**Comment by mlucool** [comment]
Thanks!

**Pull request merged**